go/vt/vttablet/tabletmanager/rpc_replication.go

```diff
@@ -54,6 +55,13 @@ func (tm *TabletManager) FullStatus(ctx context.Context) (*replicationdatapb.Ful
 	if err := tm.waitForGrantsToHaveApplied(ctx); err != nil {
 		return nil, err
 	}
+
+	// Return error if the disk is stalled or rejecting writes.
+	// Noop by default, must be enabled with the flag "enable_stalled_disk_check".
+	if tm.fsManager.IsDiskStalled() {
+		return nil, errors.New("stalled disk")
+	}
```

Comment:
I think it's better to have a boolean field or an error field here. Even if the disk is stalled, we do get all the other field information back in FullStatus that can be used. Also, an error in full status indicates to vtorc that it couldn't reach the vttablet.

Reply:
I developed this change by using `fsfreeze -f /vt` on the primary instance in a test keyspace. In my testing before making any code changes, I found that vtorc's invocation of FullStatus would timeout and return a `context deadline exceeded` error. Simple queries like `select @@global.server_id;` would hang until the filesystem was unfrozen.

We could add a boolean or error field to model this, but I think we'd want to return here either way if the check fails, and it seems a bit cleaner to me to return an error rather than a response message with mostly nil/zero values.

Reply:
I don't think it's a good architectural design to rely on the error message containing `stalled disk`. It is better to have that as a boolean field in the FullStatus output. The way the code is written now, we will always have to have the error message have `stalled disk` for backward compatibility. We can leave all other fields empty if this new boolean is set, but in my opinion we shouldn't rely on the contents of the error message.

Reply:
I agree with @GuptaManan100 here, a boolean in `FullStatus` would also allow users of `vtctldclient` to see the status using `GetFullStatus`

Reply:
Makes sense, I'll make this change 👍